### PR TITLE
Limit scope and refactor config functions

### PR DESCRIPTION
**Pull request by MalloZup**
**What type of PR is this?**
> /kind cleanup

**Special notes for your reviewer**:
split  from https://github.com/kubernetes/kubernetes/pull/73163

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by MalloZup** [comment]
@ereslibre @fabriziopandini @neolit123 @rosti CC.
 I think the bot has token a relaxation  day :relaxed:

File: cmd/kubeadm/app/cmd/join.go [hunk 1]
```diff
@@ -225,60 +225,48 @@ func NewCmdJoin(out io.Writer, joinOptions *joinOptions) *cobra.Command {
...
 
-// AddJoinConfigFlags adds join flags bound to the config to the specified flagset
-func AddJoinConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiv1beta1.JoinConfiguration) {
+// addJoinConfigFlags adds join flags bound to the config to the specified flagset
+func addJoinConfigFlags(flagSet *flag.FlagSet, cfg *kubeadmapiv1beta1.JoinConfiguration) {
 	flagSet.StringVar(
 		&cfg.NodeRegistration.Name, options.NodeName, cfg.NodeRegistration.Name,
 		`Specify the node name.`,
 	)
+	// adds file control plane flags to the specified flagset
```
Comment (rosti) [inline cmd/kubeadm/app/cmd/join.go]: "add control plane endpoint flags ..."

File: cmd/kubeadm/app/cmd/join.go [hunk 2]
```diff
@@ -225,60 +225,48 @@ func NewCmdJoin(out io.Writer, joinOptions *joinOptions) *cobra.Command {
...
 	flagSet.BoolVar(
-		&btd.UnsafeSkipCAVerification, options.TokenDiscoverySkipCAHash, false,
+		&cfg.Discovery.BootstrapToken.UnsafeSkipCAVerification, options.TokenDiscoverySkipCAHash, false,
 		"For token-based discovery, allow joining without --discovery-token-ca-cert-hash pinning.",
 	)
-}
-
-// AddJoinFileDiscoveryFlags adds file discovery flags to the specified flagset
-func AddJoinFileDiscoveryFlags(flagSet *flag.FlagSet, fd *kubeadmapiv1beta1.FileDiscovery) {
+	//	discovery via kube config file flag
```
Comment (rosti) [inline cmd/kubeadm/app/cmd/join.go]: Please, convert the tab into a space here.

**Review by rosti — COMMENTED** [review]
Thanks @MalloZup !
A couple of minor comment related changes. It looks OK and, indeed, we may have overreached ourselves with the function splits here in the past.

**Review by neolit123 — COMMENTED** [review]
/approve
/priority backlog
/sig cluster-lifecycle

**Comment by fabriziopandini** [comment]
I'm ok with this refactors, but I'm going to temporarily block any future refactor on join because the priority now is to unblock phases
/approve
I leave final lgtm to @rosti after minor comments are addressed.

File: cmd/kubeadm/app/cmd/join.go (hunk 1 above)
Comment (MalloZup) [reply cmd/kubeadm/app/cmd/join.go]: done

File: cmd/kubeadm/app/cmd/join.go (hunk 2 above)
Comment (MalloZup) [reply cmd/kubeadm/app/cmd/join.go]: i think this is a github display issue, i checked it is a space

**Review by MalloZup — COMMENTED** [review]
thx all minor comments addresses :+1:

File: cmd/kubeadm/app/cmd/join.go (hunk 2 above)
Comment (neolit123) [reply cmd/kubeadm/app/cmd/join.go]: looks like a tab in the email DIFF too.

Comment (rosti) [reply cmd/kubeadm/app/cmd/join.go]: Checked out this version of the file and opened it in vim, it's a tab. Please, fix that in order to keep the code well aligned and I'll lgtm it.

Comment (MalloZup) [reply cmd/kubeadm/app/cmd/join.go]: done @rosti thx :rocket:

**Comment by rosti** [comment]
Thanks again @MalloZup !
/lgtm

**Comment by MalloZup** [comment]
@rosti thx for awesome review. always a pleasure to commit to kubeadm. i hope to see you at next PR :+1:

**Comment by rosti** [comment]
@MalloZup can you run `./hack/update-gofmt.sh`?
/ok-to-test

**Comment by MalloZup** [comment]
@rosti should be ok now thx

**Comment by MalloZup** [comment]
seems the bot  cannot  understand the label `lgtm` :unamused: :dango:

**Pull request merged**